src/Sylius/Component/Core/Checker/Eligibility/PromotionCouponChannelEligibilityChecker.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Core\Checker\Eligibility;
+
+use Sylius\Component\Core\Model\OrderInterface as ModelOrderInterface;
+use Sylius\Component\Core\Model\Promotion as CoreModelPromotion;
+use Sylius\Component\Promotion\Checker\Eligibility\PromotionCouponEligibilityCheckerInterface;
+use Sylius\Component\Promotion\Model\PromotionCouponInterface;
+use Sylius\Component\Promotion\Model\PromotionSubjectInterface;
+
+final class PromotionCouponChannelEligibilityChecker implements PromotionCouponEligibilityCheckerInterface
+{
+    public function isEligible(PromotionSubjectInterface $promotionSubject, PromotionCouponInterface $promotionCoupon): bool
+    {
+        if (!$promotionSubject instanceof ModelOrderInterface) {
+            return true;
+        }
+
+        $promotion = $promotionCoupon->getPromotion();
+
+        if (!$promotion instanceof CoreModelPromotion) {
+            return true;
+        }
```

Comment:
why do we need those two checks?

Reply:
it is needed to get channels from `Promotion` and `PromotionSubject` as it should be instance of `Promotion` entity and `OrderInterface` to work properly

Reply:
otherwise it will show that this getChannel and getChannels methods are undefined

Reply:
in sum:
 After call we discussed that maybe We should use `Assert` or implement `Supports` method instead of this if statements